File: jar-executor-plugin/src/main/java/org/restler/CurrentProcess.java
package org.restler;

public class CurrentProcess {
    private static Process process = null;

    public static void setProcess(Process newProcess) {
        if(process != null && process.isAlive()) {
            process.destroy();
        }

        process = newProcess;
    }

    public static void killProcess() {
        if(process != null && process.isAlive()) {
            process.destroy();
        }

        process = null;
    }
}

File: jar-executor-plugin/src/main/kotlin/org/restler/CurrentProcess.kt
package org.restler

object CurrentProcess {
    private var onShutdownHook: Thread
    private var process: Process? = null

    init {
        //if maven process stopped when jar process still alive.
        onShutdownHook = Thread({CurrentProcess.killProcess()})
        Runtime.getRuntime().addShutdownHook(onShutdownHook)
    }

    fun setProcess(newProcess: Process) {
        if (process != null && process!!.isAlive) {
            process!!.destroy()
        }

        process = newProcess
    }

    fun killProcess() {
        if (process != null && process!!.isAlive) {
            process!!.destroy()
        }

        process = null
    }
}
